### PR TITLE
Fix the strides in Gluon version of ResNet v1

### PR DESCRIPTION
**Pull request by ptrendx**
## Description ##
From "Deep Residual Learning for Image Recognition" paper: Downsampling is performed by conv3_1, conv4_1, and conv5_1 with a stride of 2. In the current version it was performed in the second convolution in each block, not the first.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] For user-facing API changes, API doc string has been updated.
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- I'm not sure what to do to regenerate a pretrained model and where to put it - should I just use the example/gluon/image_classification.py script for that? @piiswrong

### CODE REVIEW

**Comment by ptrendx** [comment]
Just to clarify - this is just about version in Gluon Model ZOO - the version in `example/image-classification/symbol/resnet-v1.py` downsamples in the correct place.

**Comment by piiswrong** [comment]
The problem is we need to retrain all the pretrained models.

We plan to do this after the IO pipeline refactor, then we can test everything together

**Comment by szc1523** [comment]
@piiswrong  Is there a schedule for the  IO pipeline refactor? Thanks for your excellent work~

File: python/mxnet/gluon/model_zoo/vision/resnet.py [hunk 1]
```diff
@@ -102,10 +102,10 @@ class BottleneckV1(HybridBlock):
     def __init__(self, channels, stride, downsample=False, in_channels=0, **kwargs):
         super(BottleneckV1, self).__init__(**kwargs)
         self.body = nn.HybridSequential(prefix='')
-        self.body.add(nn.Conv2D(channels//4, kernel_size=1, strides=1))
+        self.body.add(nn.Conv2D(channels//4, kernel_size=1, strides=stride))
```
Comment (solin319) [inline python/mxnet/gluon/model_zoo/vision/resnet.py]: 1. How about add 'use_bias=False' in Conv2D? This will make resnet faster.
   The effect of bias in convolution is same as beta in the next batchnorm.
2. Does (2,2) stride set in 1*1 conv or 3*3 conv will effect the final accuracy of resnet? In 1*1 conv, (2,2) stride will loss some pixels in feature map.

**Comment by piiswrong** [comment]
@zhreshold @szha Please update the modelzoo after training with new io pipeline

**Comment by ptrendx** [comment]
Closing due to #9215

File: python/mxnet/gluon/model_zoo/vision/resnet.py (hunk 1 above)
Comment (ppwwyyxx) [reply python/mxnet/gluon/model_zoo/vision/resnet.py]: In my implementation, using stride=2 on 1x1 conv has about 0.3 lower top-1 accuracy.
